python/cudf/cudf/core/multiindex.py

```diff
@@ -775,7 +775,19 @@ def _index_and_downcast(self, result, index, index_key):
             )
 
         if isinstance(index_key, tuple):
-            result = result.set_index(index)
+            # TODO: Since index is always an Index object at this point, it
+            # should be safe to always set the index via the property and
+            # bypass the complexity of set_index for DataFrame. However, doing
+            # so results in a slight difference in the memory usage observed in
+            # test_memory_usage_multi. The likely culprit is a difference in
+            # copying semantics between the property and set_index that needs
+            # to be resolved before this change can be made. It should be only
+            # the following line once this is fixed:
+            # result.index = index
+            if isinstance(result, cudf.Series):
+                result.index = index
+            else:
+                result = result.set_index(index)
```

Comment:
The failures I observed are resolved by #10114.
```suggestion
            result.index = index
```